lib/serial_terminal.pm

```diff
@@ -232,4 +236,37 @@ sub upload_file {
       or die("Failed to finally copy file from '$tmpfilename' to '$dst'");
 }
 
+
+sub reboot {
+    my (%args) = @_;
+    $args{check} //= 1;
+    $args{console} //= testapi::current_console;
+    $args{reboot_cmd} //= 'systemctl reboot';
+    $args{timeout} //= 300;
+    $args{minimum_boot_time} = 3;
+
+    bmwqemu::log_call(%args);
+
+    my $uptime_start;
+    my $time_start = time();
+
+    die('Only root-virtio-terminal is supported') unless $args{console} eq 'root-virtio-terminal';
+    $uptime_start = script_output(q(cat /proc/uptime | awk '{print $1}')) if ($args{check});
+
+    record_info('REBOOT', "cmd: " . $args{reboot_cmd} . ($uptime_start ? "\nuptime: $uptime_start sec" : ''));
+    script_run($args{reboot_cmd}, timeout => 0);
+    reset_consoles;
+    wait_serial('[lL]ogin:', timeout => $args{timeout});
+    select_console($args{console});
+
+    my $duration = time() - $time_start;
+    if ($args{check}) {
+        my $uptime_end = script_output(q(cat /proc/uptime | awk '{print $1}'));
```

Comment:
I am not sure why you involving uptime here , you can simply measure difference between `time()` and `$time_start`

Reply:
because I will make sure, that the uptime is lower then before. This is for me the indicator, that the reboot really took place.
I didn't found something else like a boot counter or so.

Reply:
so you can simply check that `$uptime_end < $uptime_start ` ?

Reply:
and in such form I think this check should be unconditional , because if there was no uptime reset than 100% there was no reboot

Reply:
ok, then lets do it unconditional. But a check of $uptime_end < $uptime_start will not work if the reboot takes longer then the $uptime_start was. Thats why I add the duration on top of $uptime_start

Reply:
it does not matter how long reboot lasts . Let's image we have system which was running for 5 minutes . Than in [line 254](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13930/files#diff-711effe1cfae054c02b71f62a23c7a765a8aabe3f2ae80e3557e0cc0c01b27f0R254) you will get 300 . After that you trigger reboot , it could take 10 minutes or even 30 minutes you will still have 300 in variable `$uptime_start` , right ? :)  
Later in [line 264](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13930/files#diff-711effe1cfae054c02b71f62a23c7a765a8aabe3f2ae80e3557e0cc0c01b27f0R264) you will get basically time which system needed to boot which suppose to be relatively the same as before . 
The only possible race condition here that if you call this function just after system was boot and you haven't done anything else before calling it , and also there should be some "luck" with worker overloaded only during second boot. 
But I would really leave this corner case out of scope

Reply:
in other words uptime is **relative** value counted from system **start** and be designed can not be influenced by duration of reboot

Reply:
Yes exact the reason why I need to measure the duration as well. Just imagin that your call `reboot()` twice, there is quite a change to hit the case that `$uptime_end` is bigger the `$uptime_start` because of some timeout or what ever in bootup process...

Reply:
new idea - creation of folder in `/dev/shm` and check it's existence after reboot

Reply:
thx for pointing this out!